### ISSUE TITLE
Profile pic generated from public key?

### ISSUE THREAD

**Issue by jasikpark** [fraction/oasis#58]
**What's the problem you want to solved?**
![Screen Shot 2020-01-16 at 10 20 54 PM](https://user-images.githubusercontent.com/10626596/72584142-7bc20800-38ae-11ea-9951-1188fa4ec80a.png)

**Is there a solution you'd like to recommend?**

Could this profile pic be filled in with something like `color-hash`, using the public key as the hashed item? 

https://github.com/zenozeng/color-hash
https://zenozeng.github.io/color-hash/demo/

**Comment by jasikpark** [comment]
Also, the name field is `null`. Idk if that's purposeful, but I personally like the use of the first `n` characters of the public key as the defacto name.

**Comment by christianbundy** [comment]
Good idea! I'd love to merge this improvement. The only Big Decision to make is whether we want to use Sharp (which is an optional dependency because it's hard to build) or whether we want it to work for everyone. For example, if we took the first six bytes of someone's public key we could generate a 1x1 GIF with that hex color (which the browser would stretch for us). I sketched out a super quick example:

```javascript
// This is a tiny white GIF encoded with base64.
const magicBytes = "R0lGODlhAQABAIABAP///wAAACwAAAAAAQABAAACAkQBADs="
const whiteGif = Buffer.from(magicBytes, "base64")

// We need to insert our color at offset 13 to replace bytes 13, 14, and 15.
const colorOffset = 13
const colorLength = 3

// The start and end of the GIF never change, so we can repeatedly copy them.
const gifStart = whiteGif.slice(0, colorOffset)
const gifEnd = whiteGif.slice(colorOffset + colorLength)

const hexGif = input => {
  // Only convert last 6 characters, so `ffffff` and `#ffffff` are equivalent.
  const hexColor = input.slice(-6)
  const gifCopy = Buffer.from(whiteGif)
  const colorBytes = Buffer.from(hexColor, "hex")
  return Buffer.concat([gifStart, colorBytes, gifEnd])
}

// This is just an example, we wouldn't need to write to the filesystem in Oasis.
const output = hexGif("#ff00ff")
require("fs").writeFileSync("out.gif", output)
```

![out](https://user-images.githubusercontent.com/537700/72630584-7e9d1700-3907-11ea-9eb3-c0bd67fc32a4.gif)

^ It's hard to see, but I've uploaded the corresponding GIF above. ([Link](https://user-images.githubusercontent.com/537700/72630584-7e9d1700-3907-11ea-9eb3-c0bd67fc32a4.gif))

> Also, the name field is null. Idk if that's purposeful, but I personally like the use of the first n characters of the public key as the defacto name.

Works for me!

**Comment by christianbundy** [comment]
> Sharp (which is an optional dependency because it's hard to build)

Context: Sharp is only used to resize images, like profile images, before they're transferred to the browser. Lots of profile images are *huge*, so if we can resize them then we can avoid taking an 8 megapixel photo and resizing it to 40x40 pixels or whatever, but we can just fallback to the browser resizing if the `npm install` fails.
